features/mbedtls/targets/TARGET_NUVOTON/TARGET_M480/des/des_alt.c

```diff
@@ -293,36 +296,52 @@ int mbedtls_des3_crypt_ecb( mbedtls_des3_context *ctx,
  * 3DES-CBC buffer encryption/decryption
  */
 int mbedtls_des3_crypt_cbc( mbedtls_des3_context *ctx,
-                     int mode,
-                     size_t length,
-                     unsigned char iv[8],
-                     const unsigned char *input,
-                     unsigned char *output )
+                            int mode,
+                            size_t length,
+                            unsigned char iv[8],
+                            const unsigned char *input,
+                            unsigned char *output )
 {
     return mbedtls_des_docrypt(ctx->keyopt, ctx->key, mode == MBEDTLS_DES_ENCRYPT, TDES_MODE_CBC, length, iv, input, output);
 }
 #endif /* MBEDTLS_CIPHER_MODE_CBC */
 
 
 
-static int mbedtls_des_docrypt(uint16_t keyopt, uint8_t key[3][MBEDTLS_DES_KEY_SIZE], int enc, uint32_t tdes_opmode, size_t length, 
-    unsigned char iv[8], const unsigned char *input, unsigned char *output)
+static int mbedtls_des_docrypt(uint16_t keyopt, uint8_t key[3][MBEDTLS_DES_KEY_SIZE], int enc, uint32_t tdes_opmode, size_t length,
+                               unsigned char iv[8], const unsigned char *input, unsigned char *output)
 {
     if (length % 8) {
         return MBEDTLS_ERR_DES_INVALID_INPUT_LENGTH;
     }
+
+    /* DES DMA buffer has the following requirements:
+     * (1) Word-aligned buffer base address
+     * (2) 8-byte aligned buffer size
+     * (3) Located in 0x20000000-0x2FFFFFFF region
+     */
+    if ((! crypto_dma_buff_compat(dmabuf_in, MAXSIZE_DMABUF, 8)) ||
+        (! crypto_dma_buff_compat(dmabuf_out, MAXSIZE_DMABUF, 8))) {
+        error("Buffer for DES alter. DMA requires to be word-aligned and located in 0x20000000-0x2FFFFFFF region.");
+    }
+    
```

Comment:
I suggest parameter range checks at this top level function before the parameters are written to control register.

Reply:
@hanno-arm I added parameter checks in https://github.com/ARMmbed/mbed-os/pull/4925/commits/da32eed00c58d64a820e53d771f3b6e79763a499.

Reply:
@hanno-arm Continuing above, I fixed one parameter check in https://github.com/ARMmbed/mbed-os/pull/4925/commits/dbd59fd12f2b3c608ef037655ba39f2e0845556b.